internal/provider/kubernetes/gatewayclass.go

```diff
@@ -38,7 +38,7 @@ type gatewayClassReconciler struct {
 // will be pre-configured to watch for cluster-scoped GatewayClass objects with
 // a controller field that matches name.
 func newGatewayClassController(mgr manager.Manager, cfg *config.Server, su status.Updater, resources *message.ProviderResources) error {
-	resources.Initialized.Add(1)
+	resources.InitializedInfraIR.Add(1)
```

Comment:
you also need to add `InitializedXdsIR` here

Reply:
@arkodg I initially tried doing this, but the xds initialization never achieved `Done`. I will try again, push a new commit either way and maybe you can see if I'm missing something.